kedro/pipeline/node.py

```diff
@@ -198,20 +198,7 @@ def __call__(self, **kwargs) -> Dict[str, Any]:
 
     @property
     def _func_name(self):
-        if hasattr(self._func, "__name__"):
-            return self._func.__name__
-
-        name = repr(self._func)
-        if "functools.partial" in name:
-            warn(
-                "The node producing outputs `{}` is made from a `partial` function. "
-                "Partial functions do not have a `__name__` attribute: consider using "
-                "`functools.update_wrapper` for better log messages.".format(
-                    self.outputs
-                )
-            )
-            name = "<partial>"
```

Comment:
You can leave this bit here and change the condition to look like:
```
name = _get_readable..(func)
if name == "<partial>":
    warn(...)

return name
```

Otherwise if it fails on validating the inputs for a `partial` on line 540, the warning message is inaccurate because it has no knowledge of its outputs. And I would argue it's maybe cleaner to leave the warning just here, and not potentially confuse the error message.

Reply:
At one point I had checked for outputs being none and only raised if outputs were not none.  I like this better.